core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -808,15 +840,31 @@ terminate_thread_visitor(void *node, void *user_data)
 void
 wasm_cluster_terminate_all(WASMCluster *cluster)
 {
+    os_mutex_lock(&cluster->lock);
+    cluster->processing = true;
+    os_mutex_unlock(&cluster->lock);
+
     traverse_list(&cluster->exec_env_list, terminate_thread_visitor, NULL);
+
```

Comment:
does traverse_list add lock for the list? why not remove L845 and L849

Reply:
In `terminate_thread_visitor`, we will call os_thread_join to wait for other threads to exit, and the exited thread need to get `cluster->lock` for accessing the list, so we can't hold the lock during `terminate_thread_visitor`

Reply:
Got it, thanks.